src/book/02-system/01-foundations/02-build-system.mdx

```diff
@@ -566,3 +566,61 @@ target = {
+| DATA_FILES | Extra data files outside the `data` directory that is necessary for the module |
+
+Now for libraries in particular CMake needs to know where the libraries are located,
+the function `find_package()` is used to tell CMake to find the library.
+
+CMake then looks in the `cmake/Modules/` directory for files like `FindX.cmake` for the logic to find the files.
+CMake can also look for build instruction files in `lib/cmake/X/` like `XConfig.cmake` where `X` is the name of the
+library.
+An example file looks like
+
+```cmake
+include(ToolchainLibraryFinder)
+
+find_package(glib2 REQUIRED)
+
+ToolchainLibraryFinder(
+  NAME Aravis
+  HEADER arv.h
+  LIBRARY aravis-0.8
+  PATH_SUFFIX aravis-0.8
+)
+
+target_link_libraries(Aravis::Aravis INTERFACE glib2::glib2)
+```
+
+The first line includes a custom function for later use.
+We have `find_package` again, as this library has its own dependency.
+Then there is `ToolchainLibraryFinder` which is the function that finds the library.
+Finally there is `target_link_libraries` which tells CMake that we have a dependency of the library.
+`ToolchainLibraryFinder` parameters:
+
+| Parameter                | Usage                                                                                         |
+| ------------------------ | --------------------------------------------------------------------------------------------- |
+| NAME                     | The name we use for the package                                                               |
+| HEADER                   | Any header, e.g. `.h` or `.hpp`, files that define the api of the library                     |
+| LIBRARY                  | The name of the library on disk, e.g. if we have `/usr/lib/libaravis-0.8.so` it'd be `aravis` |
+| LIBRARIES                | A list instead of a single LIBRARY                                                            |
+| PATH_SUFFIX              | Add subdirectories to look for includes and libraries in the normal search directories        |
+| BINARY                   | A binary that can be used to extract the version number                                       |
+| VERSION_BINARY_ARGUMENTS | The arguments to pass to the binary that will result in a version number being printed        |
+| VERSION_FILE             | A file that contains information on the version number of the library                         |
+| VERSION_REGEX            | The regex used on the version file to extract the version number                              |
+| LINK_TYPE                | The type of linkage that the library will use: `UNKNOWN` is the default and will pick either, |
+|                          | `SHARED` will use dynamic linkage and look for a `.so` library file                           |
+|                          | `STATIC` will use static linkage and look for a `.a` library file                             |
```

Comment:
```suggestion
|      SHARED       | No arguments; use dynamic linkage and look for a `.so` library file                           |
|        STATIC     | No arguments; use static linkage and look for a `.a` library file                             |
```
maybe something like this?

Reply:
will need formatting tho

Reply:
`Unknown`, `Shared` and `Static` are options to link type.

Reply:
Oh I see, in that case put it on the one line? It looks like boolean flags here, I assumed each row of the table was a separate parameter.